src/utils/perf_log.C

```diff
@@ -111,8 +111,9 @@ void PerfLog::push (const std::string & label,
     label_c_str = non_temporary_strings[label];
   else
     {
-      char * newcopy = new char [label.size()+1];
-      strcpy(newcopy, label.c_str());
+      const std::size_t labelsizep1 = label.size()+1;
+      char * newcopy = new char [labelsizep1];
+      strncpy(newcopy, label.c_str(), labelsizep1);
```

Comment:
We include the `<cstring>` in this C file, so I would have thought `std::strncpy` would be more strictly correct here.